### ISSUE TITLE
[Down Time] Card throws 'maximum retries exceeded' error

### ISSUE THREAD

**Issue by anuraghazra** [anuraghazra/github-readme-stats#1471]
Hi everyone, I see that all personal access tokens are currently failing, thus you will see the "Maximum retries exceeded" error. 

I'm looking into the issue to figure out what caused it. 

Ideally, it should be fixed in an hour when the PATs get rest. 

# Details

## Public instance information

<img alt="Custom badge" src="https://img.shields.io/endpoint?url=https%3A%2F%2Fgithub-readme-stats-git-monitoring-github-readme-stats-team.vercel.app%2Fapi%2Fstatus%2Fup%3Ftype%3Dshields">

## Explanation

-------

Hey everyone, sorry for the recent downtimes. 

For the sake of transparency, we get a huge amount of requests per month (it's kind of mind boggling for a side project which I built just for fun). Here's the vercel dashboard statistics: 

![image](https://user-images.githubusercontent.com/35374649/194407743-9af65825-483a-49af-9e16-cac3a5ea5a50.png)

As you can see even with 73% of the responses being cached by vercel the amount of `live` requests are huge, and the bottleneck here is [Personal Access Tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) 

Each PAT have 5000 points, and based on the complexity of the GraphQL query & [resource limitations](https://docs.github.com/en/graphql/overview/resource-limitations) set by GitHub, it's actually very very low considering that amount of requests. 

To circumvent the issue, we currently have 7 PATs (thats 35k points) which sometimes get exhausted. 
Thanks to @rickstaa we are going to add few more probably bumping it up to 10 or 12, hopefully that could help alleviate the downtimes. 

> Note: That these downtimes are temporary, and only lasts for 1hour, since PATs regenerate after 1 hour

------

*So whats the idea way?*

https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-979306704
Deploy on your own vercel instance. WHY? 
It's easy to do, free & reliable. 

- It reduces loads on our vercel instance.
- Your own PAT will give you more concrete stats including private stats
- It ensures that there won't be any downtime on your profile stats because you are the only one who is gonna hit your URL

_Originally posted by @anuraghazra in https://github.com/anuraghazra/github-readme-stats/issues/2130#issuecomment-1270624232_

**Comment by jincheng9** [comment]
@anuraghazra hi, the access is still down. Do we have an expected time to fix it?

**Comment by jincheng9** [comment]
Now it's fine. Thanks.

**Comment by GabrielMendesdc** [comment]
Here still out, i need do something?

**Comment by Ari24-cb24** [comment]
Same here, any known fix to this?

**Comment by devmohit-live** [comment]
Never imagined that my profile would be that much dependent on the GitHub stats , 😁.
Hope that will be done soon, nice work man.

**Comment by anuraghazra** [comment]
Hi everyone. 

Sorry for the inconvenience. 
I will highly recommend deploying github-readme-stats on your own vercel instance to fix this issue, while I try to find a fix. Currently I'm unable to find any root cause. 

Consider deploying on your own fork. follow the guide -> https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance

Youtube video tutorial:  https://www.youtube.com/watch?v=n6d4KHSKqGk&feature=youtu.be&t=107

Also after forking make sure to open [vercel.json](https://github.com/anuraghazra/github-readme-stats/blob/ecd7d98a3c5ab9f50706458065a314b2c4f7f98e/vercel.json#L5) and change the maxDuration option to 10.

**Comment by GabrielMendesdc** [comment]
Thanks a lot! mine is already bug fixed

**Comment by anuraghazra** [comment]
> Thanks a lot! mine is already bug fixed

Yes, it's happening intermittently for some people, and for some it's not.

**Comment by elli0t43** [comment]
Its happening again.

**Comment by thyoondev** [comment]
Please fix it again. An error came up.

**Comment by elli0t43** [comment]
He is working on the fix, till then, he said to run it on our own vercel instance 

> Hi everyone.
> 
> Sorry for the inconvenience. I will highly recommend deploying github-readme-stats on your own vercel instance to fix this issue, while I try to find a fix. Currently I'm unable to find any root cause.
> 
> Consider deploying on your own fork. follow the guide -> https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance
> 
> Youtube video tutorial: https://www.youtube.com/watch?v=n6d4KHSKqGk&feature=youtu.be&t=107
> 
> Also after forking make sure to open [vercel.json](https://github.com/anuraghazra/github-readme-stats/blob/ecd7d98a3c5ab9f50706458065a314b2c4f7f98e/vercel.json#L5) and change the maxDuration option to 10.

**Comment by anuraghazra** [comment]
Added 4 new PATs, let's observe the downtime. 
For some reason the PATs are weirdly depleting super fast. not sure why. 

### *ALSO: Note that deploying on your own instance is ALWAYS a better option than relaying on ours*
1. It reduces loads on our vercel instance, we get almost 3-4Million requests per months which is hard to scale up. 
2. Your own PAT will give you more concrete stats including private stats
3.  It ensures that there won't be any downtime on your profile stats because you are the only one who is gonna hit your URL

So please I enocurage everyone to deploy on their own instance, 
**If you face any kind of issues while deployment I'm more than happy to help you out.**

**Comment by Babu-Mohammed-Izhan** [comment]
It works now !!

**Comment by Ari24-cb24** [comment]
Yeah, thanks, gotta deploy my own instance now

**Comment by debnathSD** [comment]
> Yes, it's happening intermittently for some people, and for some it's not.

Right, it is failing intermittently - checked around 6 PM IST. As @anuraghazra suggested already: `deploying github-readme-stats on your own vercel instance` would be a better option as of now.

**Comment by csharpdf** [comment]
Just so you know, I just ran into this problem too.

**Comment by KodingKhurram** [comment]
Yaah same problem here, now going forward I 'am planning to deploy this in personal vercel app :)

**Comment by jincheng9** [comment]
suggest to deploy your own instance

**Comment by codergautam** [comment]
Happening to me

**Comment by Mad880** [comment]
I'm facing the same issue as well

**Comment by michalspano** [comment]
The same issue prevails here.

**Comment by anuraghazra** [comment]
Hi folks, I think the PATs are again depleting (even though we have 6 PATs now) 

Please consider https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-979306704

**Comment by NirmalSilwal** [comment]
tagging @anuraghazra to let you know about the issue again

**Comment by StreakingMan** [comment]
It works well on this hour, and I have deploy on my own vercel, to take a little less stress on this repository's vercel instance :)

**Comment by rickstaa** [comment]
@anuraghazra Maybe we should add this to the bug template FAQ (i.e. https://github.com/anuraghazra/github-readme-stats/edit/master/.github/ISSUE_TEMPLATE/bug_report.md)?

**Comment by AneeshBose** [comment]
> Added 4 new PATs, let's observe the downtime. For some reason the PATs are weirdly depleting super fast. not sure why.
> 
> ### **ALSO: Note that deploying on your own instance is ALWAYS a better option than relaying on ours**
> 1. It reduces loads on our vercel instance, we get almost 3-4Million requests per month which is hard to scale up.
> 2. Your own PAT will give you more concrete stats including private stats
> 3. It ensures that there won't be any downtime on your profile stats because you are the only one who is gonna hit your URL
> 
> So please I enocurage everyone to deploy on their own instance, **If you face any kind of issues while deployment I'm more than happy to help you out.**

@anuraghazra  Deploying our own vercel instance requires a PAT with a maxDuration of 1-10 seconds in the free version from what I can understand. Is there any workaround to this as the next version is a paid one with $20/month. Thanks a lot once again, the cards are brilliant!

**Comment by rickstaa** [comment]
It looks like 9d2bfd9d8f00f6fc5229d0174a220ba04dbc4ee8 fixed this issue for now! 🚀 Please feel free to ping us if you experience any problems in the future.  However only comment if nobody has commented in the last three days or if you have information that could help us solve the problem. If there already is a recent comment, we are probably aware of the issue and are working on it. 🤓 Comments like "+1" and "same here" don't help anyone; they pressure the maintainer and spam their inbox. 💥

**Comment by weerdy15** [comment]
"Maxmimum retries exceeded" right now

**Comment by rickstaa** [comment]
@weerdy15 Thanks a lot for letting us know that this is still happening! I changed fixed to in-progress again.

In the meantime, please consider deploying your own Vercel instance to reduce server load (see https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance). Keeping your fork up to date is just a click of a button (see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-from-the-web-ui). This process can also be automated using a github action (see https://stackoverflow.com/a/61574295/8135687).
 
![image](https://user-images.githubusercontent.com/17570430/169291506-326ad342-1a42-438e-a522-55002d563a35.png)

**Comment by weerdy15** [comment]
@rickstaa Okay, thanks for such a great thing!

**Comment by rickstaa** [comment]
The issue should be fixed for now as we added three new Personal Access Tokens (PATs).

**Comment by ItsAleph** [comment]
> The issue should be fixed for now as we added three new Personal Access Tokens (PATs).

Can I give you mine too?

**Comment by rickstaa** [comment]
@ItsAleph Thanks a lot for offering, but that is unnecessary and not wise from a security standpoint. @anuraghazra and I will create new PAT ourselves when the current PATs run out. Are you currently experiencing problems?

**Comment by thegeek-sys** [comment]
Got the error on all my cards.
![Screenshot 2022-08-22 at 16-53-55 Editing thegeek-sys_README md at main · thegeek-sys_thegeek-sys](https://user-images.githubusercontent.com/55345633/185952073-f90f4448-622a-4e6d-925a-94e88e6145be.png)
s

**Comment by rickstaa** [comment]
As mentioned in #1986, the limits are being hit again. To resolve this, the number of PATs needs to be increased.

For people that are experiencing this issue, please consider deploying your own Vercel instance (see https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance). Doing so would decrease the load on the public server. Thanks in advance 🚀 !

**Comment by noraj** [comment]
The [@codeSTACKr](https://youtu.be/n6d4KHSKqGk?t=107) video is outdated, the vercel process changed. Same for the image guide from https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance.

And by the way just forking doesn't work anymore, during vercel deployment you will get this error:

```
Error! The value for maxDuration must be between 1 second and 10 seconds, in order to increase this limit upgrade your plan: https://vercel.com/pricing
```

So one has to edit `vercel.json` before if you plan to use the free tier:

```diff
{
  "functions": {
    "api/*.js": {
      "memory": 128,
-      "maxDuration": 30
+      "maxDuration": 10
    }
  },
  "redirects": [
    {
      "source": "/",
      "destination": "https://github.com/anuraghazra/github-readme-stats"
    }
  ]
}

```

**Comment by jcnmsg** [comment]
> So one has to edit `vercel.json` before if you plan to use the free tier

Thanks, that fixed it! I originally got the same error but the project still got created in Vercel. Simply editing the `vercel.json` file in the cloned repository and redeploying wouldn't cut it and it just kept returning **Maximum retries exceeded**. For anyone struggling with this just start over, fork this repository, edit the `vercel.json` file as explained above and then import it into a **new Vercel project** - and don't forget to add your PAT_1 environment variable.

**Comment by rickstaa** [comment]
> The [@codeSTACKr](https://youtu.be/n6d4KHSKqGk?t=107) video is outdated, the vercel process changed. Same for the image guide from https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance.
> 
> And by the way just forking doesn't work anymore, during vercel deployment you will get this error:
> 
> ```
> Error! The value for maxDuration must be between 1 second and 10 seconds, in order to increase this limit upgrade your plan: https://vercel.com/pricing
> ```
> 
> So one has to edit `vercel.json` before if you plan to use the free tier:
> 
> ```diff
> {
>   "functions": {
>     "api/*.js": {
>       "memory": 128,
> -      "maxDuration": 30
> +      "maxDuration": 10
>     }
>   },
>   "redirects": [
>     {
>       "source": "/",
>       "destination": "https://github.com/anuraghazra/github-readme-stats"
>     }
>   ]
> }
> ```

@noraj You are correct the video was done before we needed to increase the `maxDuration` parameter. Thanks for elloberating on that point! 👍🏻 However we updated the text guide some time ago to include the `maxDuration` step (see step 6). Do you think the documentation can still be improved?

**Comment by rickstaa** [comment]
I will put this test card here to see if the PATs are failing.

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=anuraghazra)](https://github.com/anuraghazra/github-readme-stats)

**Comment by rickstaa** [comment]
@anuraghazra Maybe you can add the PAT I send you over twitter? Further maybe we can update the README a bit and add a note at the top explaining that the vercel instance sometimes runs into GitHubs query limits and point out that deploying a Vercel instance is the recommended option. Additionally I think @4l1fe is currently implementing a docker container for the GRS script. If that feature is implemented I will make a small GitHub action that can be used to periodically update the cards.

**Comment by noraj** [comment]
> @noraj You are correct the video was done before we needed to increase the `maxDuration` parameter. Thanks for elaborating on that point! 👍🏻 However, we updated the text guide some time ago to include the `maxDuration` step (see step 6). Do you think the documentation can still be improved?

Yeah, for example there is no longer an import button on the dashboard but that instead 👍🏻 

![image](https://user-images.githubusercontent.com/16578570/187642129-4a26d2a7-bd00-4817-bc37-492d66686c74.png)

old import button

![image](https://user-images.githubusercontent.com/16578570/187641860-44428c2c-5515-48f9-8201-172aee739a6f.png)

this is what you got on new project creation now (looking on you organization / namespace repository or a small link to import git repo external to your oAuth app)

![image](https://user-images.githubusercontent.com/16578570/187641924-1252f446-c208-430e-bef4-cfa620a09fcf.png)

old import form

![](https://camo.githubusercontent.com/a30b3cd359d366c62498360f5f7c7eb2b679fe31a063982d44876f11b7e85892/68747470733a2f2f66696c65732e636174626f782e6d6f652f7071756239712e706e67)

etc. the interface and workflow totally changed so the video and screenshots must all be redone from scratch unfortunately

**Comment by 4l1fe** [comment]
> @anuraghazra Maybe you can add the PAT I send you over Twitter? Further, maybe we can update the README a bit and add a note at the top explaining that the Vercel instance sometimes runs into GitHub query limits and point out that deploying a Vercel instance is the recommended option. Additionally, I think @4l1fe is implementing a docker container for the GRS script. If that feature is implemented, I will make a small GitHub action that can be used to update the cards periodically to give people more options.

I'm on the way...

**Comment by rickstaa** [comment]
> > @anuraghazra Maybe you can add the PAT I send you over Twitter? Further, maybe we can update the README a bit and add a note at the top explaining that the Vercel instance sometimes runs into GitHub query limits and point out that deploying a Vercel instance is the recommended option. Additionally, I think @4l1fe is implementing a docker container for the GRS script. If that feature is implemented, I will make a small GitHub action that can be used to update the cards periodically to give people more options.
> 
> I'm on the way...

Great! 🚀 No hurry the real solution would be to add an extra PAT.

**Comment by rickstaa** [comment]
> > @noraj You are correct the video was done before we needed to increase the `maxDuration` parameter. Thanks for elaborating on that point! 👍🏻 However, we updated the text guide some time ago to include the `maxDuration` step (see step 6). Do you think the documentation can still be improved?
> 
> Yeah, for example there is no longer an import button on the dashboard but that instead 👍🏻
> 
> ![image](https://user-images.githubusercontent.com/16578570/187642129-4a26d2a7-bd00-4817-bc37-492d66686c74.png)
> 
> old import button
> 
> ![image](https://user-images.githubusercontent.com/16578570/187641860-44428c2c-5515-48f9-8201-172aee739a6f.png)
> 
> this is what you got on new project creation now (looking on you organization / namespace repository or a small link to import git repo external to your oAuth app)
> 
> ![image](https://user-images.githubusercontent.com/16578570/187641924-1252f446-c208-430e-bef4-cfa620a09fcf.png)
> 
> old import form
> 
> ![](https://camo.githubusercontent.com/a30b3cd359d366c62498360f5f7c7eb2b679fe31a063982d44876f11b7e85892/68747470733a2f2f66696c65732e636174626f782e6d6f652f7071756239712e706e67)
> 
> etc. the interface and workflow totally changed so the video and screenshots must all be redone from scratch unfortunately

Thanks for letting us know. I created #1990, so we know the problem and somebody can take it on.

**Comment by ItsAleph** [comment]
> > So one has to edit `vercel.json` before if you plan to use the free tier
> 
> Thanks, that fixed it! I originally got the same error but the project still got created in Vercel. Simply editing the `vercel.json` file in the cloned repository and redeploying wouldn't cut it and it just kept returning **Maximum retries exceeded**. For anyone struggling with this just start over, fork this repository, edit the `vercel.json` file as explained above and then import it into a **new Vercel project** - and don't forget to add your PAT_1 environment variable.

I needed to edit `vercel.json` and only *replace* PAT with a newly-generated one, then re-deploy. No need to make a new project ;D

**Comment by RafaelSolVargas** [comment]
Hi, how was it requested in https://github.com/anuraghazra/github-readme-stats/issues/1772 I'm just here to say that the Vercel limit was reached here in the last hour, more precisely between 12:00-13:00 in Brazil (GMT-3)

**Comment by rickstaa** [comment]
@RafaelSolVargas Thanks for letting us know. 👍🏻 We will add another PAT if this happens more frequently. If you want, you can deploy your own Vercel instance (see https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) to prevent this from happening.

**Comment by ctcpip** [comment]
limit is being reached currently.  I am aware of the guide on deploying an instance, but just letting you know.

**Comment by rickstaa** [comment]
@ctcpip Thanks for the heads up! We are working on it 👍🏻.

**Comment by rickstaa** [comment]
We added an extra PAT. The issue should be fixed for now.

**Comment by JacobMGEvans** [comment]
> We added an extra PAT. The issue should be fixed for now.
> 
> [![Anurag's GitHub stats](https://camo.githubusercontent.com/53b1a4c3005e1be7962d60a214d490221e3c5c976489ba1a13834d017b04d0b1/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170693f757365726e616d653d616e7572616768617a7261)](https://github.com/anuraghazra/github-readme-stats)

Its back lol

**Comment by breftejk** [comment]
It's not really.

**Comment by rickstaa** [comment]
@anuraghazra, I think we should also increase the retry number in the retry function.

https://github.com/anuraghazra/github-readme-stats/blob/94d0978d2bf7c5324513ac924cb42afe1e7774cc/src/common/retryer.js#L13

see https://github.com/anuraghazra/github-readme-stats/pull/2121.

**Comment by JacobMGEvans** [comment]
> It's not really.

The BUG is back... I should be more clear lol

**Comment by seilorjunior** [comment]
the bug is back to me

**Comment by krish4uu** [comment]
I'm also getting maximum retries exceeded!
![Screen Shot 2022-10-06 at 10 11 59 AM](https://user-images.githubusercontent.com/66221402/194335709-64f4d649-6068-43d8-83d1-2bf349fdb7b1.png)

**Comment by rickstaa** [comment]
Posting this here as a reference:

-------

Hey everyone, sorry for the recent downtimes. 

For the sake of transparency, we get a huge amount of requests per month (it's kind of mind boggling for a side project which I built just for fun). Here's the vercel dashboard statistics: 

![image](https://user-images.githubusercontent.com/35374649/194407743-9af65825-483a-49af-9e16-cac3a5ea5a50.png)

As you can see even with 73% of the responses being cached by vercel the amount of `live` requests are huge, and the bottleneck here is [Personal Access Tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) 

Each PAT have 5000 points, and based on the complexity of the GraphQL query & [resource limitations](https://docs.github.com/en/graphql/overview/resource-limitations) set by GitHub, it's actually very very low considering that amount of requests. 

To circumvent the issue, we currently have 7 PATs (thats 35k points) which sometimes get exhausted. 
Thanks to @rickstaa we are going to add few more probably bumping it up to 10 or 12, hopefully that could help alleviate the downtimes. 

> Note: That these downtimes are temporary, and only lasts for 1hour, since PATs regenerate after 1 hour

------

*So whats the idea way?*


https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-979306704
Deploy on your own vercel instance. WHY? 
It's easy to do, free & reliable. 

- It reduces loads on our vercel instance.
- Your own PAT will give you more concrete stats including private stats
- It ensures that there won't be any downtime on your profile stats because you are the only one who is gonna hit your URL

_Originally posted by @anuraghazra in https://github.com/anuraghazra/github-readme-stats/issues/2130#issuecomment-1270624232_

**Comment by rickstaa** [comment]
This bug is improved when the following PRs are merged:

- #2121: Adds support for more PATs.
- #2124: Speeds up initial card loading.
- #2159: Removes one GraphQl function call, reducing the PAT deterioration rate.
- #2177: Makes sure that errors are not cached.
- #2178: Should be checked since it looks like some PATs are expired.

**Comment by cloudybyte** [comment]
Hey there, I just encountered the same error as probably most here over the last month.
Just curious: What would prevent us/you/someone from hosting this on a VPS or sth? The request load doesn't seem that bad (10 requests/s) after all. Or is there no issue with vercel at all and its just the GitHub API quota being used up?

**Comment by rickstaa** [comment]
> Hey there, I just encountered the same error as most here over the last month. Just curious: What would prevent us/you/someone from hosting this on a VPS or sth? The request load doesn't seem that bad (10 requests/s). Or is there no issue with vertical at all, and it's just the GitHub API quota being used up?

@cloudybyte Thanks for letting us know and for your question. There exist [Vercel Limits](https://vercel.com/docs/concepts/limits/overview), but we are not close to hitting these because we using caching. The GraphQL limits cause the PAT error (see https://docs.github.com/en/graphql/overview/resource-limitations). Hosting on another provider will, therefore, not help. 

The PRs explained in https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1277550011 will improve the downtime. Apart from that, the best option is hosting your own Vercel instance since you can use your own GitHub quota (see https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance). I'm also planning on releasing a GitHub action version (see https://github.com/anuraghazra/github-readme-stats/issues/2179). This would allow for more accurate stats, no downtime and faster loading times. I am, however, waiting on some other PRs to be merged/completed before continuing this feature.

**Comment by rickstaa** [comment]
@cloudybyte When did you encounter the error? Our downtime monitor (queries every 15 min) does not show any downtime in the last seven days.

![image](https://user-images.githubusercontent.com/17570430/196466392-67e13962-b4c8-48cf-831d-fdf7535a40af.png)

**Comment by rickstaa** [comment]
Very strange. I don't have this error when I request your card at https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt.  

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt)](https://github.com/anuraghazra/github-readme-stats)
```

**Comment by cloudybyte** [comment]
> Very strange. I don't have this error when I request your card at https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt.
> 
> [![Anurag's GitHub stats](https://camo.githubusercontent.com/8ed497b712cba98b1f030025e3f4b54c57273fd5ec742b3121932dba6084fcc5/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170693f757365726e616d653d636c6f756479627974652673686f775f69636f6e733d74727565266c6f63616c653d656e267468656d653d636f62616c74)](https://github.com/anuraghazra/github-readme-stats)
> 
> ```lisp
> [![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt)](https://github.com/anuraghazra/github-readme-stats)
> ```

Thats really strange... Its working in your comment, but broken on my profile...
Thats probably sth to look into

**Comment by rickstaa** [comment]
> > @cloudybyte Thanks for letting us know and for your question. There exist [Vercel Limits](https://vercel.com/docs/concepts/limits/overview), but we are not close to hitting these because we using caching. The GraphQL limits currently cause the PAT error (see https://docs.github.com/en/graphql/overview/resource-limitations). Hosting on another provider will, therefore, not help.
> > The PRs explained in [#1471 (comment)](https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1277550011) will improve the downtime. Apart from that, the best option is hosting your own Vercel instance since you can use your own GitHub quota (see https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance). I'm also planning on releasing a GitHub action version (see #2179). This would allow for more accurate stats, no downtime and faster loading times. I am, however, waiting on some other PRs to be merged/completed before continuing this feature.
> 
> Its really not bothering me enough to set up my own vercel instance, but I wanted to improve the experience for everyone.

That's very nice of you help is always welcome! 🚀

**Comment by rickstaa** [comment]
> > Very strange. I don't have this error when I request your card at https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt.
> > [![Anurag's GitHub stats](https://camo.githubusercontent.com/8ed497b712cba98b1f030025e3f4b54c57273fd5ec742b3121932dba6084fcc5/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170693f757365726e616d653d636c6f756479627974652673686f775f69636f6e733d74727565266c6f63616c653d656e267468656d653d636f62616c74)](https://github.com/anuraghazra/github-readme-stats)
> > ```lisp
> > [![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt)](https://github.com/anuraghazra/github-readme-stats)
> > ```
> 
> Thats really strange... Its working in your comment, but broken on my profile... Thats probably sth to look into

It could be a caching problem but I thought I fixed this in https://github.com/anuraghazra/github-readme-stats/commit/4df8094e5266f28f7367552980f4e00b0c9421a9. Maybe the header is still incorrect. Could you maybe do the following troubleshooting for me?

#### Steps

1. Open https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt in a separate browser tab to see if it works.
2. Open https://github-readme-stats.vercel.app/api?username=cloudybyte&show_icons=true&locale=en&theme=cobalt. in another browser to see if it works.
3. Can you check if your browser stored cache (see https://developer.chrome.com/docs/devtools/storage/cache/).
4.

**Comment by cloudybyte** [comment]
https://user-images.githubusercontent.com/49766357/196471465-dc215627-e6dc-4df0-ac6e-33cdf0e1ba08.mp4

**Comment by cloudybyte** [comment]
1) works
2) works (tested in chromium, ff is my default)
3) there is apparently no local cache for `*.vercel.app` saved

**Comment by rickstaa** [comment]
> 1. works
> 2. works (tested in chromium, ff is my default)
> 3. there is apparently no local cache for `*.vercel.app` saved
> 
> That sounds a lot like a FF bug to me, if you prevented caching of errors by setting the correct response header...

Thanks for the information. I will check the caching headers later this week. What is a FF bug 🤓?

**Comment by rickstaa** [comment]
@cloudybyte Can you maybe check the caching headers in FF for me (see video)?

https://user-images.githubusercontent.com/17570430/196475753-feb94563-c547-4a72-9850-dd935bb32082.mp4

**Comment by rickstaa** [comment]
@cloudybyte One last thing that would help after you gave me the cache header information is to check if the problem goes away when you click the Disable Cache button. 🙏🏻 For me, that solves the problem, so the no-storage header is not working on Firefox.

![image](https://user-images.githubusercontent.com/17570430/196476666-44ee1e20-1beb-4b78-a62e-7f81d363172e.png)

**Comment by cloudybyte** [comment]
Mhm, maybe not. I tried requesting my profile with Chromium and FF in private mode and both are working.
The only issue is with my non-private-mode FF

**Comment by rickstaa** [comment]
> Ooh, I think I know where the issue is and why it only works when requesting directly: GitHub seems to cache the image itself when building the profile page. The error is probably cached on their side. That's why requesting it directly from the Vercel instance works. ![image](https://user-images.githubusercontent.com/49766357/196477044-b7685126-9dca-4e2c-bade-3843068faf09.png)

Great detective work! Could be the issue. I will check if the GitHub CDN respects the `no-store` header.

**Comment by cloudybyte** [comment]
It doesn't look like GitHub being the issue, based on [my other findings](https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1282597331).

**Comment by rickstaa** [comment]
> Based on [my other findings](https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1282597331), it doesn't look like GitHub is the issue.

So it just works if you use the GitHub CDN URL in any other browser?

**Comment by cloudybyte** [comment]
Yeah

**Comment by cloudybyte** [comment]
Yea

**Comment by rickstaa** [comment]
> Yea

It looks like a particular Vercel + Firefox error. The card shown in https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1282585613 is coming from the Vercel cache (time 15:30).

![image](https://user-images.githubusercontent.com/17570430/196480847-f39f80ab-c77a-4636-a9de-d360a325bfdf.png)

**Comment by Kentakoong** [comment]
<img width="838" alt="Screen Shot 2022-10-18 at 23 20 18" src="https://user-images.githubusercontent.com/35761701/196487055-b1fdeb3a-a6ad-41db-895a-4a4dcf842256.png">

mine's not showing the GH Stats and the Top Langs

**Comment by cloudybyte** [comment]
Can you try fetching the stats directly using this url: `https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong` and tell us which browser u're using? (Your profile is broken on my side, but fetching directly works fine...)

**Comment by rickstaa** [comment]
> <img alt="Screen Shot 2022-10-18 at 23 20 18" width="838" src="https://user-images.githubusercontent.com/35761701/196487055-b1fdeb3a-a6ad-41db-895a-4a4dcf842256.png">
> 
> mine's not showing the GH Stats and the Top Langs
> 
> ps: I'm using the img tag so I guess it's a caching issue like said in the upper comments!

@Kentakoong, now it doesn't seem to be a caching issue anymore. The PATs quota for this hour has simply been used up. To decrease this occurrence, #2121 has to be merged, and some PATs have to be replaced (see #2121). I will check back in an hour.

**Comment by rickstaa** [comment]
> Can you try fetching the stats directly using this url: `https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong` and tell us which browser u're using? (Your profile is broken on my side, but fetching directly works fine...)

You are right! Fetching directly still works, so maybe the PATs still have quota left, and this is a cached response on the server side that is still being shown from the GitHub CDN. Very strange problem.

**Your syntax markdown**

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong)](https://github.com/anuraghazra/github-readme-stats)
```

**Your syntax img**

<div align="center">
<img src="https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong" height="150" alt="ghstats"  />
</div>

```md
<div align="center">
<img src="https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong" height="150" alt="ghstats"  />
</div>
```

**Your syntax in markdown while breaking the cache with &safdsaf=adsfsaf**


[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong&safdsaf=adsfsaf)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong&safdsaf=adsfsaf)](https://github.com/anuraghazra/github-readme-stats)
```

**Your syntax img while breaking the cache with &safdsaf=adsfsaf**

<div align="center">
<img src="https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong&safdsaf=adsfsaf" height="150" alt="ghstats"  />
</div>

```md
<div align="center">
<img src="https://github-readme-stats.vercel.app/api?hide_rank=true&show_icons=true&include_all_commits=false&count_private=true&disable_animations=false&theme=dark&locale=en&hide_border=true&custom_title=Github&nbsp;Stats&username=Kentakoong&safdsaf=adsfsaf" height="150" alt="ghstats"  />
</div>
```

**Comment by rickstaa** [comment]
@cloudybyte and @
Kentakoong  It Looks like the problem at the current time only is present when the `<IMG>` tag is used (see https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-1282690404). This, however, is, I think, a coincidence and could as well have been the other way around. 

I think the GitHub CDN could still cause this caching problem. This behaviour can be seen in action if I used the cached image URL directly:

<div align="center">
<img src="https://camo.githubusercontent.com/bdc4b7001ae4db64d42b0567e58c40584162aff3cd12f250869298dafec66d78/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170693f686964655f72616e6b3d747275652673686f775f69636f6e733d7472756526696e636c7564655f616c6c5f636f6d6d6974733d66616c736526636f756e745f707269766174653d747275652664697361626c655f616e696d6174696f6e733d66616c7365267468656d653d6461726b266c6f63616c653d656e26686964655f626f726465723d7472756526637573746f6d5f7469746c653d476974687562c2a0537461747326757365726e616d653d4b656e74616b6f6f6e67" height="150" alt="ghstats"  />
</div>

```html
<div align="center">
<img src="https://camo.githubusercontent.com/bdc4b7001ae4db64d42b0567e58c40584162aff3cd12f250869298dafec66d78/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170693f686964655f72616e6b3d747275652673686f775f69636f6e733d7472756526696e636c7564655f616c6c5f636f6d6d6974733d66616c736526636f756e745f707269766174653d747275652664697361626c655f616e696d6174696f6e733d66616c7365267468656d653d6461726b266c6f63616c653d656e26686964655f626f726465723d7472756526637573746f6d5f7469746c653d476974687562c2a0537461747326757365726e616d653d4b656e74616b6f6f6e67" height="150" alt="ghstats"  />
</div>
```

To be sure, I asked a question in the [Vercel repository](https://github.com/vercel/vercel/discussions/8741).

**Comment by rickstaa** [comment]
@cloudybyte I think I found the problem. The GitHub CDN needs the `no-cache` (see https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls#an-image-that-changed-recently-is-not-updating). I updated the cache using the `curl -X PURGE https://camo.githubusercontent.com/bdc4b7001ae4db64d42b0567e58c40584162aff3cd12f250869298dafec66d78/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170693f686964655f72616e6b3d747275652673686f775f69636f6e733d7472756526696e636c7564655f616c6c5f636f6d6d6974733d66616c736526636f756e745f707269766174653d747275652664697361626c655f616e696d6174696f6e733d66616c7365267468656d653d6461726b266c6f63616c653d656e26686964655f626f726465723d7472756526637573746f6d5f7469746c653d476974687562c2a0537461747326757365726e616d653d4b656e74616b6f6f6e67` command and now your profile is working again.

**Comment by cloudybyte** [comment]
Mhm, that sounds weird. Thanks for fixing it tho. 
hmu if u need help or anything

**Comment by rickstaa** [comment]
Yea strange behaviour indeed the GitHub CDN should also respect `no-store` header. I think 98f9045f987112bb3d4f1a0160f3e0d93ef5bd52 will fix the issues. Please feel free to comment below if it occurs again.

**Comment by HADMARINE** [comment]
Just informing that the server seems to be down for a while
![image](https://user-images.githubusercontent.com/46421618/197804876-22b25c3a-3c57-4f7c-81ad-4d02d97fd36d.png)

**Comment by rickstaa** [comment]
> This bug is improved when the following PRs are merged:
> 
> * [fix: fix retry max-out bug #2121](https://github.com/anuraghazra/github-readme-stats/pull/2121): Adds support for more PATs.
> * [feat: improve card loading speed #2124](https://github.com/anuraghazra/github-readme-stats/pull/2124): Speeds up initial card loading.
> * [feat: enable multi page star fetching for private vercel instances #2159](https://github.com/anuraghazra/github-readme-stats/pull/2159): Removes one GraphQl function call, reducing the PAT deterioration rate.
> * [fix: prevent totalCommitsFetch error result from being cached #2177](https://github.com/anuraghazra/github-readme-stats/pull/2177): Makes sure that errors are not cached.
> * [feat: add PAT monitoring functions #2178](https://github.com/anuraghazra/github-readme-stats/pull/2178): Should be checked since it looks like some PATs are expired.

@HADMARINE, thanks for letting us know the instance was indeed down for about 10 minutes. @anuraghazra, can you look at the issues above if you find the time? Especially #2178. It seems that only one of our PATs is still working. 🤔

**Comment by jesterb0206** [comment]
![image](https://user-images.githubusercontent.com/109996512/203603417-ba4ad067-46d0-4816-ab0e-43d4d46d566e.png)

@anuraghazra @rickstaa I'm receiving it this morning

**Comment by rickstaa** [comment]
We are aware of the issue and actively monitoring the PATs thanks for letting us know. 👍🏻

**Comment by dhiegodias** [comment]
Down here as well :(
![image](https://user-images.githubusercontent.com/67385608/204560354-744c8ac1-2163-48e4-a14e-1b558ff4bc09.png)

Thanks guys!

**Comment by showierdata9978** [comment]
You need to create a GitHub personal access token, and put it in a env var
called PAT_1

**Comment by showierdata9978** [comment]
if you have ever used git on the cmdline its basically the same thing

here`s the url to make a token: https://github.com/settings/tokens
how to add an env var to  Vercel: https://vercel.com/docs/concepts/projects/environment-variables

**Comment by dginovker** [comment]
@rickstaa Is there a place we can provide you PAT tokens?

**Comment by mrtnvgr** [comment]
@rickstaa @anuraghazra this appeared again

**Comment by rickstaa** [comment]
> @rickstaa @anuraghazra this appeared again ![](https://camo.githubusercontent.com/73c2b0007b3d46ba9d67b9c0e6663b4e20334ac82949c313afd2bf40ffb6589b/68747470733a2f2f692e696d6775722e636f6d2f5a374265326e552e706e67)

Thanks for letting us know. 👍

**Comment by Ddhruv-IOT** [comment]
It's down again
![image](https://user-images.githubusercontent.com/54676859/212042833-55a7f18d-c2a7-42b8-927d-4a38459d9a2b.png)

**Comment by rickstaa** [comment]
> It's down again ![image](https://user-images.githubusercontent.com/54676859/212042833-55a7f18d-c2a7-42b8-927d-4a38459d9a2b.png)

That looks like a different error I will check.

**Comment by rickstaa** [comment]
Duplicate of #2380.

**Comment by showierdata9978** [comment]
How is this a dupe, this is #1471

**Comment by rickstaa** [comment]
> How is this a dupe, this is #1471

Good point I accidentally closed the main issue. 😅

**Comment by FaraiMajor** [comment]
https://github-readme-stats-sigma-five.vercel.app

this fixed it for me. added sigma five

**Comment by RedHeadphone** [comment]
Deploying own Vercel works for few hours then again `Maximum retries exceeded` error is returned from the API, my guess is that github has updated some limitation on usage of PAT

**Comment by rickstaa** [comment]
> retries

@RedHeadphone This is strange for your personal Vercel instance it should be more than enough. You can check how many requests you get in the Vercel interface. 🤔 Github gives users 5000 points, and I never had problems.

**Comment by RedHeadphone** [comment]
> @RedHeadphone This is strange for your personal Vercel instance it should be more than enough. You can check how many requests you get in the Vercel interface. 🤔 Github gives users 5000 points, and I never had problems. Maybe you are still referring the public Vercel instance in your README.

The Vercel Analytics doesn't seem to show number of requests. And no I am referring to my new vercel instance in README.
One strange thing to be noted is that the a PAT which was giving `Maximum retries exceeded` in one of my vercel instance, using same PAT on other vercel instance seems to be working.

I have mentioned about same in [this issue](https://github.com/anuraghazra/github-readme-stats/issues/2402)

![image](https://user-images.githubusercontent.com/55500003/212600473-bbddccb9-4c82-4a37-85c8-948ada72237d.png)

**Comment by rickstaa** [comment]
> > @RedHeadphone This is strange for your personal Vercel instance it should be more than enough. You can check how many requests you get in the Vercel interface. 🤔 Github gives users 5000 points, and I never had problems. Maybe you are still referring the public Vercel instance in your README.
> 
> The Vercel Analytics doesn't seem to show number of requests. And no I am referring to my new vercel instance in README. One strange thing to be noted is that the a PAT which was giving `Maximum retries exceeded` in one of my vercel instance, using same PAT on other vercel instance seems to be working.
> 
> I have mentioned about same in [this issue](https://github.com/anuraghazra/github-readme-stats/issues/2402)
> 
> ![image](https://user-images.githubusercontent.com/55500003/212600473-bbddccb9-4c82-4a37-85c8-948ada72237d.png)

You can find this information under the Usage tab in the general Vercel dashboard. I think the best option is to create a new PAD. 👍

![image](https://user-images.githubusercontent.com/17570430/212663056-e5a77d4f-ae20-4b83-be32-25d19949e12c.png)

**Comment by ParoaPe** [comment]
> 
Thanks for this solution, this is working for me (18 January 2023). 

This section: https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance and this
YouTube video tutorial: https://www.youtube.com/watch?v=n6d4KHSKqGk&feature=youtu.be&t=107 combined have resolved my problems

**Comment by rayennecandido** [comment]
> @rickstaa @anuraghazraisso apareceu de novo ![](https://camo.githubusercontent.com/73c2b0007b3d46ba9d67b9c0e6663b4e20334ac82949c313afd2bf40ffb6589b/68747470733a2f2f692e696d6775722e636f6d2f5a374265326e552e706e67)
https://www.youtube.com/watch?v=n6d4KHSKqGk&t=107s

**Comment by jamesthesnake** [comment]
Seems to have happened again.

**Comment by filipporomani** [comment]
@anuraghazra yup, it's happening again!

**Comment by ParoaPe** [comment]
> I think it's happening again

This is working for me (18 January 2023).

This section: https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance and this
YouTube video tutorial: https://www.youtube.com/watch?v=n6d4KHSKqGk&feature=youtu.be&t=107 combined have resolved my problems

**Comment by jventura1738** [comment]
> I think it's happening again

It's working now.

**Comment by mehmetaligumusler** [comment]
Seems to have happened again.

**Comment by huanmeng-qwq** [comment]
Seems to have happened again.

**Comment by WeismannS** [comment]
Happened again today!

**Comment by ennioVisco** [comment]
Today as well:
<img width="602" alt="image" src="https://user-images.githubusercontent.com/740025/226614463-26326487-5b70-4f05-b48c-7049248ad683.png">

**Comment by kms0219kms** [comment]
It is happening again, in 00:39 25/03/2023 in GMT+7.
![image](https://user-images.githubusercontent.com/67222970/227600268-53a263ff-1005-40a8-8f4c-2a3dca0621c7.png)

**Comment by rickstaa** [comment]
> It is happening again, in 00:39 25/03/2023 in GMT+7. ![image](https://user-images.githubusercontent.com/67222970/227600268-53a263ff-1005-40a8-8f4c-2a3dca0621c7.png)

Thanks for letting us know. This is a known limitation of GitHub's GraphQL API (see https://github.com/anuraghazra/github-readme-stats/issues/1471). I have sent @anuraghazra a new PAT. Meanwhile, [deploy your own Vercel instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own) or try [the experimental GitHub action](https://github.com/marketplace/actions/grs-action).

**Comment by showierdata9978** [comment]
You could swap PAT's every 30 minutes, if they are close to the limit, with a backup one for 30 minutes while the original one refreshes its use limit

**Comment by rickstaa** [comment]
> You could swap PAT's every 30 minutes, if they are close to the limit, with a backup one for 30 minutes while the original one refreshes its use limit

Thanks for the suggestion. We already use something similar in the code base:

https://github.com/anuraghazra/github-readme-stats/blob/32998295b7ef9742ea0168213807d7e8ec5c72a6/src/fetchers/stats-fetcher.js#L146C9-L170

**Comment by jventura1738** [comment]
It's back again

**Comment by rjscdev** [comment]
![image](https://user-images.githubusercontent.com/88680048/231846286-443181ee-39d3-4b28-a70e-7bd7daa24bb3.png)
again

**Comment by davina708** [comment]
<img width="604" alt="스크린샷 2023-04-27 오전 1 26 44" src="https://user-images.githubusercontent.com/111004706/234641028-1efdeb35-7aec-4cba-974e-8dc6947f8e26.png">

Can anyone help me with this issue?

**Comment by AbhayaShankar** [comment]
Something went wrong! file an issue at https://tiny.one/readme-stats
Maximum retries exceeded Please add an env variable called PAT_1 with your github token in vercel
Happened again

**Comment by AbhayaShankar** [comment]
![image](https://user-images.githubusercontent.com/113185442/235726108-64dbcb7a-a9f1-43ff-943f-5379d2456292.png)

**Comment by NamitBhutani** [comment]
is there some down time going on? Because I hosted on my own on vercel, but I still get this issue of Maximum tries exceeded.

**Comment by NamitBhutani** [comment]
> > is there some down time going on? Because I hosted on my own on vercel, but I still get this issue of Maximum tries exceeded.
> 
> how popular is your profile, if you get over 1000 visits per hour (?) it would do that

No, not that famous ...

**Comment by RobThree** [comment]
Experiencing the same:

![image](https://github.com/anuraghazra/github-readme-stats/assets/607888/a86e9089-d49a-49a6-ad41-33df7c40a786)

**Comment by KraXen72** [comment]
happening to me as well with PAT_1

**Comment by ev1025** [comment]
I have created a project on Vercel and deployed it, but the GitHub token keeps getting deleted, causing this error to occur repeatedly. What should I do?

**Comment by Himan10** [comment]
@ev1025 @KraXen72 hey guys, sorry for the tag but I just encountered this issue as well, just want to know how did you resolve it ? It says adding an environment variable PAT_1 with the GitHub token but I am not sure what its usage ?

**Comment by ev1025** [comment]
@Himan10 
I think you'd better look at this on https://github.com/anuraghazra/github-readme-stats/issues/1471#issuecomment-979201779 .

**Comment by Vallek** [comment]
Same error here with PAT_1. I'm pretty sure i don't have a lot of viewers on my profile😅

**Comment by AshwanthramKL** [comment]
![image](https://github.com/anuraghazra/github-readme-stats/assets/71945723/201a9ae8-2d01-4171-9006-5085582fafe7)

Still exists

**Comment by jdevstatic** [comment]
back then, there was the suggestion 

`https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app`

now, I'm restoring it to the original 

`https://github-readme-stats.vercel.app`

and it's working again

**Comment by FahadBinHussain** [comment]
![Screenshot_2023-09-05-20-50-05-630_com kiwibrowser browser](https://github.com/anuraghazra/github-readme-stats/assets/75292632/b82373c7-9847-4e51-9bc1-e47b561d2e58)
 
Please fix this ASAP

**Comment by qwerty541** [comment]
> ![Screenshot_2023-09-05-20-50-05-630_com kiwibrowser browser](https://user-images.githubusercontent.com/75292632/265738201-b82373c7-9847-4e51-9bc1-e47b561d2e58.png)
> 
> Please fix this ASAP

@FahadBinHussain Sometimes you can see these error because our public instance is being rate-limited by underlying GitHub API. You can [deploy your own instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own) using your own GitHub API token in order to fix this or wait until the public instance is longer rate-limited.

**Comment by FahadBinHussain** [comment]
> > ![Screenshot_2023-09-05-20-50-05-630_com kiwibrowser browser](https://user-images.githubusercontent.com/75292632/265738201-b82373c7-9847-4e51-9bc1-e47b561d2e58.png)
> > Please fix this ASAP
> 
> @FahadBinHussain Sometimes you can see these error because our public instance is being rate-limited by underlying GitHub API. You can [deploy your own instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own) using your own GitHub API token in order to fix this or wait until the public instance is longer rate-limited.

Yeah, I have read that. However, I am currently facing certain circumstances which prevent me from deploying my own instance. That's why I am seeking assistance here to rectify the issues with the public one.

**Comment by qwerty541** [comment]
> > > ![Screenshot_2023-09-05-20-50-05-630_com kiwibrowser browser](https://user-images.githubusercontent.com/75292632/265738201-b82373c7-9847-4e51-9bc1-e47b561d2e58.png)
> > > Please fix this ASAP
> > 
> > 
> > @FahadBinHussain Sometimes you can see these error because our public instance is being rate-limited by underlying GitHub API. You can [deploy your own instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own) using your own GitHub API token in order to fix this or wait until the public instance is longer rate-limited.
> 
> Yeah, I have read that. However, I am currently facing certain circumstances which prevent me from deploying my own instance. That's why I am seeking assistance here to rectify the issues with the public one.
> 
> But, thank you. Since the issue has been resolved.

Unfortunately, there is not much we can do with the public instance. We are limited by the number of requests from the GitHub API. In theory, we can increase the already large number of API tokens that the public instance uses, but this will not solve the problem. Firstly, only the owner of the repository can do this, and secondly, the number of users is constantly growing and very soon we will again face the same problem. Currently the best way to avoid this issue is deploying own instance, if you have any idea how we can resolve it with public one feel free to share.

**Comment by Stk-79** [comment]
> > > > ![Screenshot_2023-09-05-20-50-05-630_com kiwibrowser browser](https://user-images.githubusercontent.com/75292632/265738201-b82373c7-9847-4e51-9bc1-e47b561d2e58.png)
> > > > Please fix this ASAP
> > > 
> > > 
> > > @FahadBinHussain Sometimes you can see these error because our public instance is being rate-limited by underlying GitHub API. You can [deploy your own instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own) using your own GitHub API token in order to fix this or wait until the public instance is longer rate-limited.
> > 
> > 
> > Yeah, I have read that. However, I am currently facing certain circumstances which prevent me from deploying my own instance. That's why I am seeking assistance here to rectify the issues with the public one.
> > But, thank you. Since the issue has been resolved.
> 
> Unfortunately, there is not much we can do with the public instance. We are limited by the number of requests from the GitHub API. In theory, we can increase the already large number of API tokens that the public instance uses, but this will not solve the problem. Firstly, only the owner of the repository can do this, and secondly, the number of users is constantly growing and very soon we will again face the same problem. Currently the best way to avoid this issue is deploying own instance, if you have any idea how we can resolve it with public one feel free to share.

Read everything available to fix this issues along with this two #2402 #2404 ..Hope it will fire up again. Thanks

**Comment by FredHappyface** [comment]
Hi thank you for this awesome project!

Just a thought: is it worth setting the default cache to a higher value? Such as 12 hours or so? I say this as for many users there's not going to be much difference in stats over this period and hopefully it further reduces strain on the PATs. Possibly keeping the 4 hours as a minimum so users can opt in to current cache times 

Cheers 🙂

**Comment by qwerty541** [comment]
> Hi thank you for this awesome project!
> 
> Just a thought: is it worth setting the default cache to a higher value? Such as 12 hours or so? I say this as for many users there's not going to be much difference in stats over this period and hopefully it further reduces strain on the PATs. Possibly keeping the 4 hours as a minimum so users can opt in to current cache times
> 
> Cheers 🙂

Hi, @FredHappyface! I like your idea and think that increasing the minimum caching period on public instance is something we should do, thanks for sharing. If someone want his stats to be reloaded more oftern he will be able to deploy his own instance and set environment variable `CACHE_SECOND`.

@rickstaa What do you think about this? Are you up to approve pull request with such change if i implement it?

**Comment by rickstaa** [comment]
> > Hi thank you for this awesome project!
> > Just a thought: is it worth setting the default cache to a higher value? Such as 12 hours or so? I say this as for many users there's not going to be much difference in stats over this period and hopefully it further reduces strain on the PATs. Possibly keeping the 4 hours as a minimum so users can opt in to current cache times
> > Cheers 🙂
> 
> Hi, @FredHappyface! I like your idea and think that increasing the minimum caching period on public instance is something we should do, thanks for sharing. If someone want his stats to be reloaded more oftern he will be able to deploy his own instance and set environment variable `CACHE_SECOND`.
> 
> @rickstaa What do you think about this? Are you up to approve pull request with such change if i implement it?


I'm comfortable with extending the cache period to 12 hours since it will reduce the frequency of creating Personal Access Tokens (PATs) that need to be sent to @anuraghazra. However, one concern I have is that this change might lead to more users reporting issues about their GitHub cards not updating promptly. 

Do you think the recent addition I made to the readme (https://github.com/anuraghazra/github-readme-stats/commit/3e66189c44fca0ff89ea975b7f71f84fbdf512ab) effectively communicates the use of our caching mechanism?

**Comment by qwerty541** [comment]
> > > Hi thank you for this awesome project!
> > > Just a thought: is it worth setting the default cache to a higher value? Such as 12 hours or so? I say this as for many users there's not going to be much difference in stats over this period and hopefully it further reduces strain on the PATs. Possibly keeping the 4 hours as a minimum so users can opt in to current cache times
> > > Cheers 🙂
> > 
> > 
> > Hi, @FredHappyface! I like your idea and think that increasing the minimum caching period on public instance is something we should do, thanks for sharing. If someone want his stats to be reloaded more oftern he will be able to deploy his own instance and set environment variable `CACHE_SECOND`.
> > @rickstaa What do you think about this? Are you up to approve pull request with such change if i implement it?
> 
> I'm comfortable with extending the cache period to 12 hours since it will reduce the frequency of creating Personal Access Tokens (PATs) that need to be sent to @anuraghazra. However, one concern I have is that this change might lead to more users reporting issues about their GitHub cards not updating promptly.
> 
> Do you think the recent addition I made to the readme ([3e66189](https://github.com/anuraghazra/github-readme-stats/commit/3e66189c44fca0ff89ea975b7f71f84fbdf512ab)) effectively communicates the use of our caching mechanism?

I have opened pull request https://github.com/anuraghazra/github-readme-stats/pull/3242 with extending default cache time to 8 housr for the beginning and little improvements to documentation with explanation of our caching mechanism. Please check it when you have free time.

**Comment by ashishagarwal2023** [comment]
@anuraghazra I'd recommend you to simply do this: make multiple servers from the main server.

You can simply take the main server and host 3 different/other servers for same thing. Main server will go to the server with least-used time, so that it can give you 3x speed

**Comment by qwerty541** [comment]
> @anuraghazra I'd recommend you to simply do this: make multiple servers from the main server.
> 
> You can simply take the main server and host 3 different/other servers for same thing. Main server will go to the server with least-used time, so that it can give you 3x speed

Hey, @ashishagarwal2023! Thanks for your input. Looks like that you a bit misunderstood the problem. We do not experience thr problems with Vercel servers productivity or slow response time. Our problem is 5000 points limitation per GitHub API token, so when it hits the limit the server statrs showing the error. Currently our team trying to handle this problem by the following actions:

- Iterating through several GitHub API tokens when trying to featch users data, so one of them should work.
- Caching response for 6 hours, to avoid using API points on each request.
- Motivating users to deploy their own instances by providing them special features like ability to set custom cache time.

**Comment by pwbriggs** [comment]
[shields.io](https://shields.io) ran into a similar issue a while back, and GitHub support recommended a clever solution at https://github.com/badges/shields/issues/529#issuecomment-228605811. You could consider a similar solution, though it might be difficult to implement (and the requests would still be running on your Vercel instance). Something worth considering, though 🤷

**Comment by F18-Maverick** [comment]
Hi, I'm having the same issue, but I didn't use the official url. I forked the project and deploy the project by my own vercel. How can I resolve this problem?

By the way, the reason why I deployed it on my own Vercel account is because it can't be loaded while using the official url, and after I changed the url to my own vercel url, it can be loaded, but then I encountered this error. So could you please help me about this problem? (my own vercel host: [github-readme-stats-f18.vercel.app](https://github-readme-stats-f18.vercel.app/))

<img width="613" height="155" alt="Image" src="https://github.com/user-attachments/assets/bf8d2b6b-6fae-4a0c-a2fd-a2d80f14278b" />

The image shows the state after I changed the URL host.

**Comment by pwbriggs** [comment]
@F18-Maverick sounds like you need to configure your token on your instance. Check steps 9-10 in the [deploy your own instructions](https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#on-vercel) on the readme. If you followed all those instructions, please open a new issue because it's likely unrelated to this one. Thanks!

**Comment by qwerty541** [comment]
> Hi, I'm having the same issue, but I didn't use the official url. I forked the project and deploy the project by my own vercel. How can I resolve this problem?
> 
> By the way, the reason why I deployed it on my own Vercel account is because it can't be loaded while using the official url, and after I changed the url to my own vercel url, it can be loaded, but then I encountered this error. So could you please help me about this problem? (my own vercel host: [github-readme-stats-f18.vercel.app](https://github-readme-stats-f18.vercel.app/))
> 
> <img alt="Image" width="613" height="155" src="https://private-user-images.githubusercontent.com/140802139/520355464-bf8d2b6b-6fae-4a0c-a2fd-a2d80f14278b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ0NTc4MjIsIm5iZiI6MTc2NDQ1NzUyMiwicGF0aCI6Ii8xNDA4MDIxMzkvNTIwMzU1NDY0LWJmOGQyYjZiLTZmYWUtNGEwYy1hMmZkLWEyZDgwZjE0Mjc4Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyOVQyMzA1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OTgxOWRlYWVjYjcyNjZhNTk2Mzg4NzQyYTk0ZWMxMDk4ZjYyOGUxYzJiNjBjYWQ2YTVjNmQyMjc3MjI1NWNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.z9LxN67dwgA_V6QB8WEeXl0Zrls16tbRvMpcYIwQWOQ">
> The image shows the state after I changed the URL host.

Hi @F18-Maverick 

The error on your card reflecting that there is no required environment variable "PAT_1" with your GitHub API token. So you have to add this as show in our deployment guideline inside readme file.

In case you're sure that environment variable already exists with correct name and value, then you can try to redeploy the instance, as far as I remember the changes in environment variables in applied only to new deployments on Vercel.

**Comment by F18-Maverick** [comment]
Okay, thanks @pwbriggs and @qwerty541. I'll try it later!

**Comment by RedaArrous** [comment]
how to deploy it?

**Comment by pwbriggs** [comment]
Hey @RedaArrous! Check out the ["Deploy on your own" section](https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#deploy-on-your-own) of this repo's readme. If something goes wrong while following those instructions, please search the issues to see if somebody's had your problem before, and if not, open a new issue with an error report. Issues with deploying your own instance are likely off-topic for this issue. Thanks!

**Comment by RedaArrous** [comment]
@pwbriggs thanks for the polite response man .
